python/lsst/ts/scheduler/utils/test/feature_scheduler_sim.py

```diff
@@ -0,0 +1,218 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["FeatureSchedulerSim"]
+
+import logging
+import types
+
+from astropy.time import Time
+
+from lsst.ts.scheduler.driver import FeatureScheduler
+
+from lsst.ts.observatory.model import ObservatoryModel
+from lsst.ts.observatory.model import ObservatoryState
+
+from lsst.ts.dateloc import ObservatoryLocation
+
+from lsst.ts.astrosky.model import AstronomicalSkyModel
+
+from rubin_sim.site_models.seeingModel import SeeingModel
+from rubin_sim.site_models.cloudModel import CloudModel
+from rubin_sim.site_models.downtimeModel import DowntimeModel
+
+
+class FeatureSchedulerSim:
+    """Utility class to simulate observations with the FeatureScheduler.
+
+    This class is used mostly for unit testing.
+    """
+
+    def __init__(self, log: logging.Logger) -> None:
+
+        self.log = log.getChild(__name__)
+
+        self.start_time = Time(59853.983, format="mjd", scale="tai")
+        # Step in time when there is no target (in seconds).
+        self.no_target_time_step = 120.0
+
+        self.raw_telemetry = dict()
+
+        self.raw_telemetry["timeHandler"] = None
+        self.raw_telemetry["scheduled_targets"] = []
+        self.raw_telemetry["observing_queue"] = []
+        self.raw_telemetry["observatoryState"] = None
+        self.raw_telemetry["bulkCloud"] = 0.0
+        self.raw_telemetry["seeing"] = 1.19
+
+        self.models = dict()
+
+        self.models["location"] = ObservatoryLocation()
+        self.models["location"].for_lsst()
+
+        self.models["observatory_model"] = ObservatoryModel(self.models["location"])
+        self.models["observatory_model"].configure_from_module()
+
+        self.models["observatory_model"].update_state(self.start_time.unix)
+
+        self.models["observatory_state"] = ObservatoryState()
+        self.models["observatory_state"].set(
+            self.models["observatory_model"].current_state
+        )
+
+        self.models["sky"] = AstronomicalSkyModel(self.models["location"])
+        self.models["seeing"] = SeeingModel()
+        self.models["cloud"] = CloudModel()
+        self.models["downtime"] = DowntimeModel()
+
+        self.driver = FeatureScheduler(
+            models=self.models, raw_telemetry=self.raw_telemetry
+        )
+
+        self.config = types.SimpleNamespace(
+            driver_configuration=dict(
+                scheduler_config="",
+                force=True,
+                default_observing_script_name="standard_visit.py",
+                default_observing_script_is_standard=True,
+                stop_tracking_observing_script_name="stop_tracking.py",
+                stop_tracking_observing_script_is_standard=True,
+            )
+        )
+
+        self.files_to_delete = []
+
+    def make_scheduler_configuration(
```

Comment:
Please add doc strings to this and all following methods. The names of the methods are well chosen and give an idea of what the methods do but they also contain jargon related to the Scheduler that not all readers may be familiar with.